be/src/exec/pipeline/crossjoin/cross_join_left_operator.cpp

```diff
@@ -0,0 +1,419 @@
+        if (src_col->is_constant()) {
+            // current can't reach here
+            auto* const_col = vectorized::ColumnHelper::as_raw_column<vectorized::ConstColumn>(src_col);
+            // repeat the constant value for copy_number times
+            _buf_selective.assign(row_count, 0);
+            dest_col->append_selective(*const_col->data_column(), &_buf_selective[0], 0, row_count);
+        } else {
+            dest_col->append(*src_col.get(), start_row, row_count);
+        }
+    } else {
+        if (src_col->is_constant()) {
+            // current can't reach here
+            dest_col->append_nulls(row_count);
+        } else {
+            dest_col->append(*src_col.get(), start_row, row_count);
+        }
+    }
+}
+
+void CrossJoinLeftOperator::_copy_build_rows_with_index_base_build(vectorized::ColumnPtr& dest_col,
+                                                                   vectorized::ColumnPtr& src_col, size_t start_row,
+                                                                   size_t row_count) {
+    if (!src_col->is_nullable()) {
+        if (src_col->is_constant()) {
+            // current can't reach here
+            auto* const_col = vectorized::ColumnHelper::as_raw_column<vectorized::ConstColumn>(src_col);
+            // repeat the constant value for copy_number times
+            _buf_selective.assign(row_count, 0);
+            dest_col->append_selective(*const_col->data_column(), &_buf_selective[0], 0, row_count);
+        } else {
+            dest_col->append_value_multiple_times(*src_col.get(), start_row, row_count);
+        }
+    } else {
+        if (src_col->is_constant()) {
+            // current can't reach here
+            dest_col->append_nulls(row_count);
+        } else {
+            dest_col->append_value_multiple_times(*src_col.get(), start_row, row_count);
+        }
+    }
+}
+
+/* 
+ * This algorithm is the same as that CrossJoinNode, 
+ * and pull_chunk, need_input, push_chunk is splited from CrossJoinNode's get_next.
+ */
+StatusOr<vectorized::ChunkPtr> CrossJoinLeftOperator::pull_chunk(RuntimeState* state) {
+    vectorized::ChunkPtr chunk = nullptr;
+    // If right table is empty, so we just return empty chunk.
+    if (_total_build_rows > 0) {
+        for (;;) {
+            if (chunk == nullptr) {
+                // we need a valid probe chunk to initialize the new chunk.
+                _init_chunk(&chunk);
+            }
+
+            // need row_count to fill in chunk.
+            size_t row_count = config::vector_chunk_size - chunk->num_rows();
+
+            // means we have scan all chunks of right tables.
+            // we should scan all remain rows of right table.
+            // once _probe_chunk_index == _probe_chunk->num_rows() is true,
+            // this condition will always true for this _probe_chunk,
+            // Until _probe_chunk be done.
+            if (_probe_chunk_index == _probe_chunk->num_rows()) {
```

Comment:
size_t num_rows = _probe_chunk->num_rows()

change the following _probe_chunk_index with num_rows, cause it looks a little bit confusing